python/ql/src/experimental/Security/CWE-730/RegexInjection.ql

```diff
@@ -0,0 +1,29 @@
+/**
+ * @name Regular expression injection
+ * @description User input should not be used in regular expressions without first being escaped,
+ *              otherwise a malicious user may be able to inject an expression that could require
+ *              exponential time on certain inputs.
+ * @kind path-problem
+ * @problem.severity error
+ * @id py/regex-injection
+ * @tags security
+ *       external/cwe/cwe-730
+ *       external/cwe/cwe-400
+ */
+
+// determine precision above
+import python
+import experimental.semmle.python.security.injection.RegexInjection
+import DataFlow::PathGraph
+
+from
+  RegexInjectionFlowConfig config, DataFlow::PathNode source, DataFlow::PathNode sink,
+  RegexInjectionSink castedSink, Attribute methodAttribute
+where
+  config.hasFlowPath(source, sink) and
+  castedSink = sink.getNode() and
+  methodAttribute = castedSink.getRegexMethod()
+select sink.getNode(), source, sink,
+  "$@ regular expression is constructed from a $@ and executed by $@.", sink.getNode(), "This",
+  source.getNode(), "user-provided value", methodAttribute,
+  castedSink.getRegexModule() + "." + methodAttribute.getName()
```

Comment:
I might rename `castedSink` to `regexInjectionSink`, since it is more important that it is of that class than that it has been cast. (Incidentally, I believe the past tense of "cast" is also "cast", so it should be `castSink` if you insist on that shorter naming scheme.)

Reply:
Sounds good, thanks for the suggestion! https://github.com/github/codeql/pull/5442/commits/20b532ec5e3b57790cf20ea7ebc587b9fcadc10d :)